gammapy/analysis/config.py

```diff
@@ -0,0 +1,189 @@
+        json_encoders = {
+            AngleType: lambda v: f"{v.value} {v.unit}",
+            EnergyType: lambda v: f"{v.value} {v.unit}",
+            TimeType: lambda v: f"{v.value}",
+        }
+
+    @classmethod
+    def from_yaml(cls, filename):
+        config = read_yaml(filename)
+        return Config(**config)
+
+    def to_yaml(self):
+        return yaml.dump(yaml.safe_load(self.json()))
+
+    def update_from_dict(self, other):
+        data = deep_update(self.dict(exclude_defaults=True), other.dict(exclude_defaults=True))
+        return Config(**data)
+
+
+class Skydir(GammapyBaseModel):
+    frame: FrameEnum = FrameEnum.icrs
+    lon: AngleType = AngleType("83.633 deg")
+    lat: AngleType = AngleType("22.014 deg")
+
+
+class EnergyAxis(GammapyBaseModel):
+    min: EnergyType = EnergyType("0.1 TeV")
+    max: EnergyType = EnergyType("10 TeV")
+    nbins: int = 30
+
+
+class SpatialCircleRange(GammapyBaseModel):
+    frame: FrameEnum = FrameEnum.icrs
+    lon: AngleType = AngleType("83.633 deg")
+    lat: AngleType = AngleType("22.014 deg")
+    radius: AngleType = AngleType("0.1 deg")
+
+
+class EnergyRange(GammapyBaseModel):
+    min: EnergyType = EnergyType("0.1 TeV")
+    max: EnergyType = EnergyType("10 TeV")
+
+
+class TimeRange(GammapyBaseModel):
+    start: TimeType = None
+    stop: TimeType = None
+
+
+class FluxPoints(GammapyBaseModel):
+    energy: EnergyAxis = EnergyAxis()
+
+
+class Fit(GammapyBaseModel):
+    fit_range: EnergyRange = EnergyRange()
+
+
+class Background(GammapyBaseModel):
+    method: BackgroundMethodEnum = BackgroundMethodEnum.reflected
+    exclusion: FilePath = None
+
+
+class Axes(GammapyBaseModel):
+    energy: EnergyAxis = EnergyAxis()
+    energy_true: EnergyAxis = EnergyAxis()
+
+
+class Selection(GammapyBaseModel):
+    offset_max: AngleType = AngleType("2.5 deg")
+
+
+class Fov(GammapyBaseModel):
+    width: AngleType = AngleType("5 deg")
+    height: AngleType = AngleType("5 deg")
+
+
+class Wcs(GammapyBaseModel):
+    skydir: Skydir = Skydir()
+    binsize: AngleType = AngleType("0.1 deg")
+    fov: Fov = Fov()
+    binsize_irf: AngleType = AngleType("0.1 deg")
+    margin_irf: AngleType = AngleType("0.1 deg")
+
+
+class Geom(GammapyBaseModel):
+    wcs: Wcs = Wcs()
+    selection: Selection = Selection()
+    axes: Axes = Axes()
+
+
+class Datasets(GammapyBaseModel):
+    type: str = "1d"
+    stack: bool = True
+    geom: Geom = Geom()
+    background: Background = Background()
+    onregion: SpatialCircleRange = SpatialCircleRange()
+    containment_correction: bool = True
+    psf_kernel_radius: AngleType = AngleType("0.6 deg")
+
+
+class Data(GammapyBaseModel):
+    datastore: Path = "$GAMMAPY_DATA/hess-dl3-dr1/"
+    obs_ids: List[int] = []
+    obs_file: FilePath = None
+    obs_cone: SpatialCircleRange = SpatialCircleRange()
+    obs_time: TimeRange = TimeRange()
+
+    @validator("datastore")
+    def datastore_exists(cls, v):
+        return make_path(v)
```

Comment:
Remove this validator?
It doesn't check if the path is really present on disk.
Probably we shouldn't until that path is accessed, because someone might write a script where they generate that path only after the loading the config file?
Generally I don't think creating a config should hit the disk and check stuff there.

Reply:
Yes.
Note we also have some `FilePath` types that checks the existence of the file. 
https://pydantic-docs.helpmanual.io/usage/types/#pydantic-types

It is not the case for the datastore folder. Anyway, the functions should not have that name since it does not check existence of datastore folder, it just expands the env vars on setting (not reading), which I guess it should not be the task of the model.